doc/source/text.rst

```diff
@@ -168,28 +168,34 @@ Extracting Substrings
 
 .. _text.extract:
 
-The method ``extract`` (introduced in version 0.13) accepts `regular expressions
-<https://docs.python.org/2/library/re.html>`__ with match groups. Extracting a
-regular expression with one group returns a Series of strings.
+.. versionadded:: 0.13.0
 
-.. ipython:: python
+.. warning::
+
+   In version 0.18.0, ``extract`` gained the ``expand`` argument. When
+   ``expand=False`` (default) it returns a ``Series``, ``Index``, or
+   ``DataFrame``, depending on the subject and regular expression
```

Comment:
About the PR of @sinhrks (#10103),
```
+        * If False, return Series/Index.
```
I don't think that makes sense for a regex with multiple groups (unless we want to raise `ValueError` when `expand=False`).

I think it makes more sense to do as @jreback proposed (have `None` just be an alias for the default, with a warning).